servers/physics_3d/godot_body_pair_3d.cpp

```diff
@@ -251,7 +239,10 @@ bool GodotBodyPair3D::setup(real_t p_step) {
 
 	offset_B = B->get_transform().get_origin() - A->get_transform().get_origin();
 
-	validate_contacts();
+	// reset contacts, keep prior frame's contacts as cache for warm start.
+	std::swap(prior_contacts, contacts);
```

Comment:
I believe the engine provided `SWAP` is preferred over the standard library one.

Reply:
SWAP doesn't seem to support arrays. Compiler errors occur if I change it to SWAP:
`
typedefs.h(135): error C3074: an array cannot be initialized with a parenthesized initializer
typedefs.h(136): error C3863: array type 'GodotBodyContact3D::Contact [4]' is not assignable
`